app/src/main/kotlin/codes/chrishorner/socketweather/home/ForecastUi.kt

```diff
@@ -196,6 +203,47 @@ private fun SectionEntry(@DrawableRes iconRes: Int, iconDesc: String, content: S
   }
 }
 
+@Composable
+private fun Description(
+  description: Description,
+  onToggle: (Boolean) -> Unit
+) {
+  val (text, expandable, expanded) = description
+
+  Column(
+    modifier = Modifier
+      .padding(start = 16.dp, end = 16.dp, bottom = 16.dp)
+      .fillMaxWidth()
+      .clickable(
+        enabled = expandable,
+        onClick = { onToggle(!expanded) },
+        onClickLabel = stringResource(
+          if (expanded)
+            R.string.home_expandedDescriptionOnClickLabel
+          else
+            R.string.home_collapsedDescriptionOnClickLabel
+        )
+      )
+  ) {
+    Text(
+      text = text,
+      style = MaterialTheme.typography.body1,
+      modifier = Modifier.animateContentSize()
+    )
+
+    if (expandable) {
+      val rotation: Float by animateFloatAsState(if (expanded) 0F else -360F)
+      Icon(
+        imageVector = if (expanded) Icons.Rounded.ExpandLess else Icons.Rounded.ExpandMore,
```

Comment:
We can simplify this by not swapping out the icon.

```suggestion
      val rotation: Float by animateFloatAsState(if (expanded) 180f else 0f)
      Icon(
        imageVector = Icons.Rounded.ExpandMore,
```

Reply:
Nice one!